test/jest/sequencer.ts

```diff
@@ -0,0 +1,45 @@
+import Sequencer, { ShardOptions } from "@jest/test-sequencer";
+import { Test } from "@jest/test-result";
+import slash = require("slash");
+import * as crypto from "crypto";
+import * as path from "path";
+
+export default class CustomSequencer extends Sequencer {
+  /**
+   * Select tests for shard requested via --shard=shardIndex/shardCount
+   * Sharding is applied before sorting
+   * Based on the default Jest sharder except it avoids empty shards
+   * https://github.com/facebook/jest/blob/main/packages/jest-test-sequencer/src/index.ts#L40
+   * See https://github.com/facebook/jest/issues/13027 for more context
+   */
+  shard(tests: Array<Test>, options: ShardOptions): Array<Test> | Promise<Array<Test>> {
```

Comment:
Basic idea is:

Where m is buckets and n is the number of tests

`n/m` = initial size of bucket, add 1 to the first `n%m `buckets

I _think_ this is O(n/m) :P

There's my interview practice for the year.